vike/node/plugin/plugins/importUserCode/v1-design/getVikeConfig.ts

```diff
@@ -1170,6 +1184,43 @@ async function findPlusFiles(
   return plusFiles
 }
 
+async function gitLsFiles(userRootDir: string, outDir: string) {
+  const cmd = [
+    'git ls-files',
+    ...scriptFileExtensionList.map((ext) => `"**/+*.${ext}"`),
+    '--exclude="**/node_modules/**"',
+    `--exclude="${outDir}/**"`,
+    // Allow:
+    // ```
+    // +Page.js
+    // +Page.telefunc.js
+    // ```
+    '--exclude="**/*.telefunc.*"',
+    // --others lists untracked files only (but using .gitignore because --exclude-standard)
+    // --cached adds the tracked files to the output
+    '--others --cached --exclude-standard'
+  ].join(' ')
+
+  let stdout: string
+  try {
+    const res = await execA(cmd, { cwd: userRootDir })
+    stdout = res.stdout
+  } catch (err) {
+    if ((err as Error).message.includes('not a git repository')) return null
+    throw err
+  }
+
+  let files = stdout.split('\n').filter(Boolean)
+
+  // --exclude only applies to untracked files
+  assert(!outDir.startsWith('/'))
+  files = files.filter(
+    (line) => !line.includes('node_modules/') && !line.includes('.telefunc.') && !line.startsWith(`${outDir}/`)
+  )
```

Comment:
Aren't we excluding all this already on lines 1191-1198 anyway?

Reply:
Those `--exclude` only applies to the untracked files. So if, for some reason `outDir` is committed and tracked, we need to filter it here.

Reply:
Yep. (And FYI that's why I decided to not factor out the patterns between `fast-glob` and `git ls-files`.)

I improved the comment to make it clearer why we repeat the exclusion logic.